File: ios/Familienkalender/Repositories/AIRepository.swift
import Foundation

final class AIRepository: Sendable {
    private let api: APIClient

    init(apiClient: APIClient) { self.api = apiClient }

    func getAvailableRecipes(weekStart: String) async throws -> AvailableRecipesResponse {
        let query = [URLQueryItem(name: "week_start", value: weekStart)]
        return try await api.get(path: "\(Endpoints.AI.base)/available-recipes", queryItems: query)
    }

    func generateMealPlan(_ request: GenerateMealPlanRequest) async throws -> PreviewMealPlanResponse {
        try await api.post(path: "\(Endpoints.AI.base)/generate-meal-plan", body: request)
    }

    func confirmMealPlan(_ request: ConfirmMealPlanRequest) async throws -> ConfirmMealPlanResponse {
        try await api.post(path: "\(Endpoints.AI.base)/confirm-meal-plan", body: request)
    }

    func undoMealPlan(_ request: UndoMealPlanRequest) async throws -> [String: AnyCodableValue] {
        try await api.post(path: "\(Endpoints.AI.base)/undo-meal-plan", body: request)
    }

    func voiceCommand(text: String) async throws -> VoiceCommandResponse {
        let body = VoiceCommandRequest(text: text)
        return try await api.post(path: "\(Endpoints.AI.base)/voice-command", body: body)
    }
}

File: obsolete/android/app/src/main/java/de/familienkalender/app/data/repository/AiRepository.kt
package de.familienkalender.app.data.repository

import de.familienkalender.app.data.remote.api.AiApi
import de.familienkalender.app.data.remote.dto.*

class AiRepository(
    private val api: AiApi
) {

    suspend fun getAvailableRecipes(weekStart: String): Result<AvailableRecipesResponse> {
        return try {
            Result.success(api.getAvailableRecipes(weekStart))
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun generateMealPlan(request: GenerateMealPlanRequest): Result<PreviewMealPlanResponse> {
        return try {
            Result.success(api.generateMealPlan(request))
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun confirmMealPlan(request: ConfirmMealPlanRequest): Result<ConfirmMealPlanResponse> {
        return try {
            Result.success(api.confirmMealPlan(request))
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun undoMealPlan(mealIds: List<Int>): Result<Map<String, Any>> {
        return try {
            Result.success(api.undoMealPlan(UndoMealPlanRequest(mealIds)))
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun voiceCommand(text: String): Result<VoiceCommandResponse> {
        return try {
            Result.success(api.voiceCommand(VoiceCommandRequest(text)))
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}
